tests/collector/test_local_collector.py

```diff
@@ -26,6 +28,34 @@ def create_local_collector(tmp_path_factory: TempPathFactory) -> tuple[LocalColl
     return LocalCollector(path, indexer), path
 
 
+class WaitForAsyncFunctionCall:
+    def __init__(self, func: Callable):
+        """
+        Wrapper around a function that is being watched.
+
+        :param func: Function to wait for.
+        """
+        self.func = func
+        self.fut = get_running_loop().create_future()
+
+    async def wrap(self, *args, **kwargs):  # type: ignore
+        """Method to pass as side_effect to patch(.object)."""
+        ret = await self.func(*args, **kwargs)
+        self.fut.set_result(True)
+        return ret
```

Comment:
Falls beim Aufruf von `self.func` eine Exception auftritt, wartet `self.wait_for_fn_call` so lange, bis es zum `asyncio.TimeoutError` kommt.

```suggestion
    async def wrap(self, *args: Any, **kwargs: Any) -> Any:
        """Method to pass as side_effect to patch(.object)."""
        try:
            ret = await self.func(*args, **kwargs)
            self.fut.set_result(True)
            return ret
        except Exception as e:  # pylint: disable=broad-except
            self.fut.set_exception(e)
```